MSBot/src/msbot-connect-file.ts

```diff
@@ -0,0 +1,68 @@
+import * as program from 'commander';
+import * as chalk from 'chalk';
+import * as fs from 'fs-extra';
+import * as getStdin from 'get-stdin';
+import * as path from 'path';
+import { BotConfig, ServiceType } from './BotConfig';
+import { Enumerable, List, Dictionary } from 'linq-collections';
+import { uuidValidate } from './utils';
+import { IConnectedService, ILuisService, IDispatchService, IAzureBotService, IBotConfig, IEndpointService, IQnAService, IFileService } from './schema';
+
+program.Command.prototype.unknownOption = function (flag: any) {
+    console.error(chalk.default.redBright(`Unknown arguments: ${flag}`));
+    program.help();
+};
+
+interface ConnectFileArgs extends IFileService {
+    bot: string;
+    secret: string;
+}
+
+program
+    .name("msbot connect file <path>")
+    .description('Connect a file to the bot')
+    .option('-b, --bot <path>', "path to bot file.  If omitted, local folder will look for a .bot file")
+    .option('--secret <secret>', 'bot file secret password for encrypting service secrets')
+    .action((filePath, actions) => {
+        if (filePath)
```

Comment:
Nice!

Reply:
I wonder though, while file makes sense as dispatch generic source, would it make sense as bot service?  But we need this since dispatch will only have ids...